app/controllers/application_controller.rb

```diff
@@ -25,7 +28,22 @@ class ApplicationController < ActionController::Base
   end
 
   def authenticated?
-    session["authentication"].present?
+    authenticated_via_account? || session["authentication"].present?
+  end
+
+  def authenticated_via_account?
+    return false unless govuk_account_auth_enabled?
+    return false if account_session_header.blank?
+    return @authenticated_via_account unless @authenticated_via_account.nil?
+
+    api_response = GdsApi.email_alert_api.link_subscriber_to_govuk_account(govuk_account_session: account_session_header)
```

Comment:
I think it's a little confusing to see `link_subscriber_to_govuk_account` in a method that's called on every request. I understand why you've used it - anyone signed into the email alert frontend with their GOV.UK account should already have their accounts linked, so this will just return the subscriber object. 

I think `authenticate_subscriber_by_govuk_account` will do the same job here and be a lot clearer for future readers of this code. Plus we wouldn't run the risk of linking someone's accounts without informing them if there's an edge case we've not considered

Reply:
> anyone signed into the email alert frontend with their GOV.UK account should already have their accounts linked, so this will just return the subscriber object.

Actually that's not the case, with this if you sign in to the email management frontend *without* having linked your account, it will be done in the background without you being asked if that's what you want.  Though you will get an email.

I raised this in our QA walkthrough, and this is desired behaviour: we think it will affect sufficiently few users that we don't need to make a separate screen for this.

Reply:
Ah ok then 👍